### ISSUE TITLE
ssh issues with block : many issues, can't give a name

### ISSUE THREAD

**Issue by deltasata** [warpdotdev/Warp#5250]
### Dupe Check

- [X] I have [searched Warp SSH issues](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ASSH) and there are no duplicates

### Operating System

Linux

### Operating System and Version

Ubuntu 22.04

### Local Shell Version

5.2.21(1)-release

### Remote Shell Version

5.2.12(1)-release

### Warp Version

v0.2024.08.20.08.02.stable_00

### Have you tried commenting out my system & user rc files?

- [X] Yes

### Screenshots

![Screenshot from 2024-08-22 22-21-35](https://github.com/user-attachments/assets/0f72b114-51a1-4a00-b616-e5affeefc224)

Multiple weird issues. But I will explain only the ones that force me not to use warp.
1. First of all, I have to disable SSH wrapper because of other issues (it can't remember my 2FA validation even with the correct ssh config file). This is just a statement, not a question, let's not get into the details. 
2. With ssh wrapper disabled, when I log into the remote host, my cursor becomes thick, not blinking.
3.  Right click menu changes, no paste option.
4. Most annoying, after I paste something by middle click, nothing can be done by keyboard until I click on the block. That is possibly because, after middle click paste the block is no longer highlighted.
5. With ssh wrapper enabled, in first remote host I don't see the above issues. But I face the same problems with the nested ssh sessions. Before you question, nested ssh is becoming the norm in the science community (when using large clusters). 

What surprises me, all other terminals never cause a single problem executing these simplest ssh things! I blame the block implementation. I wish one day somebody convinces you to ditch this block implementation. Or, the other terminal copies your aesthetics and the auto-complete implementation. 

### Include shell xtrace output

This is not necessary since this happens to all the remote hosts I logged into. SSH connection is established fine.

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore) - linear-label:e7dfaa84-5fdb-4a00-b754-d8912da923fa

None

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Hi @deltasata thanks for letting us know. Warp has an SSH Wrapper that enables features (like the blocks, prompt, completions, etc.) on remote machines requiring the default prompt on the remote machine to be either `bash` or `zsh`.

As a workaround you can add `command ssh` to your `Settings > Subshells > Added commands` menu, then run `command ssh <user@server>` to connect to remote session, this will attempt to enable Warp features as a subshell. https://docs.warp.dev/features/ssh#ssh-wrapper-fails

You can also try disabling the SSH Wrapper by navigating to `Settings > Features`. Blocks will stop working but with it disabling, SSH should work as normal.

If those don't work, please let us know.

**Comment by deltasata** [comment]
As I explained before, with SSH wrapper or command ssh, the problem reappears for the 2nd nested ssh sessions (I must need this).
So lets focus on ssh wrapper disabled. Yes, I explained the problem above, plenty of weird stuff. Please read it carefully.

**Comment by dannyneira** [comment]
Yes when using the `command ssh` method, you should be able to have a nested SSH session with Warp features enabled. So just run `command ssh` again for the new server and then "Warpify" the session with `CTRL-i`. I've attached a demo of this. 

https://github.com/user-attachments/assets/45d9d2f8-0b15-4c8a-993a-10470ee0b73f

That being said we understand there are issues with SSH which we are actively working to improve.

**Comment by deltasata** [comment]
1.5 half years.. still no improvement .. Nobody even copied your aesthetics and auto-complete features!!! Strange world!
